dashboard/api.py

```diff
@@ -367,3 +380,127 @@ def format_courserun_for_dashboard(course_run, status_for_user, certificate=None
         # TODO: here goes the logic to pull the current grade  # pylint: disable=fixme
         pass
     return formatted_run
+
+
+def get_student_enrollments(user, edx_client):
```

Comment:
right, so if Enrollment and Certificate inherit from the same parent class, then that would be a function that we would implement differently for Enrollment and Certificate. these two classes follow extraordinarily similar patterns, and the differences between them would be covered by implementing some specific function differently in each class. a lot of this code is copy/pasted (comments from lines 425 to 427 indicate 'certificate' and the function is pulling enrollments). that by itself is a pretty big indicator that we are missing some abstraction. again, i'm happy to push this to a different story since that is a bigger change and this issue is time-sensitive